### PR TITLE
store: always log retry summary when SNAPD_DEBUG is set

### PR DESCRIPTION
**Pull request by mvo5**
This should help to track down: https://travis-ci.org/snapcore/snapd/jobs/195605347#L1619

Here we try only once except that we got a io.EOF error that should trigger a retry. The improved logging hopefully has clues why no retry happend.

### CODE REVIEW

File: store/store.go [hunk 1]
```diff
@@ -601,6 +601,19 @@ func maybeLogRetryAttempt(reqOptions *requestOptions, attempt *retry.Attempt, st
...
+func maybeLogRetrySummary(reqOptions *requestOptions, attempt *retry.Attempt, resp *http.Response, err error, startTime time.Time) {
+	if osutil.GetenvBool("SNAPPY_TESTING") || attempt.Count() > 1 {
+		var status string
+		delta := time.Since(startTime) / time.Millisecond
+		if err != nil {
+			status = err.Error()
+		} else if resp != nil {
+			status = fmt.Sprintf("%d", resp.StatusCode)
+		}
+		logger.Debugf("The retry loop for %s finished after %d retries, delta time=%v ms, status: %s", reqOptions.URL, attempt.Count(), delta, status)
```
Comment (pedronis) [inline store/store.go]: s/delta/elapsed/ ?

File: store/store.go [hunk 2]
```diff
@@ -601,6 +601,19 @@ func maybeLogRetryAttempt(reqOptions *requestOptions, attempt *retry.Attempt, st
 
 }
 
+func maybeLogRetrySummary(reqOptions *requestOptions, attempt *retry.Attempt, resp *http.Response, err error, startTime time.Time) {
+	if osutil.GetenvBool("SNAPPY_TESTING") || attempt.Count() > 1 {
```
Comment (niemeyer) [inline store/store.go]: Sounds good. Can this be brought more inline with the other variables we have?  SNAPD_DEBUG?

File: store/retry.go [hunk 3]
```diff
@@ -42,24 +42,28 @@ var defaultRetryStrategy = retry.LimitCount(5, retry.LimitTime(33*time.Second,
 ))
 
 func maybeLogRetryAttempt(url string, attempt *retry.Attempt, startTime time.Time) {
-	if osutil.GetenvBool("SNAPPY_TESTING") || attempt.Count() > 1 {
+	if osutil.GetenvBool("SNAPD_DEBUG") || attempt.Count() > 1 {
 		logger.Debugf("Retyring %s, attempt %d, elapsed time=%v", url, attempt.Count(), time.Since(startTime))
```
Comment (pedronis) [inline store/retry.go]: already there, but there's typo: s/Retyring/Retrying/

**Comment by stolowski** [comment]
LGTM, thank you!

**Pull request merged**